V2

stefanpante opened the pull request.

stefanpante committed:
refactor && add sessionStore functionality && sso

stefanpante committed:
fix bugs & update inner workings

stefanpante committed:
updated readm

stefanpante committed:
add trust proxy

stefanpante committed:
fix authentication methods table

stefanpante committed:
fix authentication methods table

stefanpante committed:
update readme and use hooks

stefanpante committed:
update readme and use hooks

stefanpante committed:
- update readme
- update tests

stefanpante committed:
remove unused dependencies & files

stefanpante committed:
2.0.0-0

Jan-Bart commented on examples/basic-mongo/package.json:
```diff
@@ -9,7 +9,7 @@
   "license": "ISC",
   "dependencies": {
-    "@digipolis/auth": "^1.1.0",
+    "@digipolis/auth": "git+https://github.com/digipolisantwerp/auth_module_nodejs.git",
```
Waarom is dit niet de gepubliceerde versie?

stefanpante replied:
omdat deze versie nog niet gepublished is.

Jan-Bart commented on readme.md:
````diff
@@ -1,32 +1,134 @@
-```js
-app.use(require('@digipolis/auth')(app, configuration));
-```
+You should use `express-session` in your application to enable session storage.
````
Misschien handig om achter `express-session` een linkje te leggen naar de juiste module:
https://www.npmjs.com/package/express-session

Jan-Bart commented on src/index.js:
```diff
@@ -0,0 +1,31 @@
+];
+
+function validateConfig(config) {
+
```
whiteline

Jan-Bart commented on test/loginCallback.js:
```diff
@@ -0,0 +1,210 @@
+    });
+
+    res.on('end', () => {
+      // assert(res.redirect.calledWith('/'));
```
oeps?

Jan-Bart commented on test/sso.js:
```diff
@@ -0,0 +1,222 @@
+
+
+describe('test sso middleware', function onDescribe() {
+  // nockGetSession({ssoKey: 'fakessokey', payload: emptySessions})
```
Mag weg.

Jan-Bart commented on examples/basic-mongo/docker-compose.yml:
```diff
@@ -6,6 +6,8 @@ services:
       - PORT=2000
       - MONGO_CONNECTIONSTRING=mongodb://mongo:27017/basic-mongo-auth
+      - CLIENT_SECRET=
+      - CLIENT_ID
```
Misschien handig om in te vullen met dummy code?

Jan-Bart commented:
Ik zou ook node v6 uit de test-builds halen en expliciet vermelden in de README dat we dit niet meer ondersteunen.

frankdsm commented on test/isLoggedin.js:
```diff
@@ -1,13 +1,14 @@
 const user = require('./mocks/user.json');
 
-describe('test #isLoggedin', function onDescribe() {
+describe('GET /isLoggedin', function onDescribe() {
```
Lowercase urls pls

frankdsm reviewed (commented):
Only es module imports pls

stefanpante commented on src/controller.js:
```diff
@@ -0,0 +1,274 @@
+    }
+  
+    const newToken = await service.refresh(token);
+    req.session = Object.assign(req.session, { [tokenKey]: newToken });
```
Maakt rest operator er geen nieuw object van, ik kopieer hier de property op de req.session
Misschien is de assignment dan niet echt nodig.

stefanpante committed:
change build plan to support down till 8

stefanpante committed:
fix casing

stefanpante committed:
2.0.0-1

stefanpante committed:
wrong default causes problems (should be array instead of string)

PrinceAnsong committed:
Use default authentication method in SSO redirect URL for low sessions (#25)

stefanpante committed:
2.0.0-2

PrinceAnsong committed:
Fix typo in highest assurance level search loop (#26)

* Use default authentication method in SSO redirect URL for low sessions

* Fix typo in loop while searching for highest assurance level

* Remove node 8 from travis config

PrinceAnsong committed:
Fix naming of authentication method param during logout

Jan-Bart committed:
Merge pull request #27 from PrinceAnsong/bugfix/fix-renamed-authentication-method

Fix naming of authentication method param during logout

Jan-Bart committed:
Version bump

Jan-Bart committed:
2.0.0-4

PrinceAnsong committed:
Delete sessions after successful logout instead of before logout

Deleting a session for an itsme authentication before the logout flow is started causes the flow to terminate in the consent app before the user is redirected to the itsme logout service. So when the user attempts to log in again, when the user reaches the itsme server to log in, the pre-existing session is used to automatically log the user in.

See: https://jira.antwerpen.be/browse/AUTH-394?focusedCommentId=227930&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-227930

Jan-Bart committed:
Merge pull request #28 from PrinceAnsong/bugfix/delete-sessions-after-logout

Delete sessions after successful logout instead of before logout

Jan-Bart committed:
2.0.0-5

stefanpante committed:
hinted login is workiung (#32)

* hinted login is workiung

* 2.1.0-0

* fix access token expiryu

* 2.1.0-1

* make it possible to disable upgrading the user's assurancelevel

* 2.1.0-2 (#35)

* update & fix default query param

stefanpante committed:
2.1.0

stefanpante committed:
add logger

stefanpante committed:
2.1.1-0

stefanpante committed:
add debug info

stefanpante committed:
2.1.1-1

stefanpante committed:
update logging

stefanpante committed:
2.1.1-2

stefanpante committed:
add additional login, change itsme to high

stefanpante committed:
2.1.1-3

stefanpante committed:
revert itsme to high

stefanpante committed:
2.1.1-4

stefanpante committed:
add logs to sso middleware

stefanpante committed:
give users the ability to seit  save_consent to false

Give user the ability to override the save_consent default

stefanpante committed:
2.1.1

stefanpante committed:
2.2.0

stefanpante committed:
add force auth param

stefanpante committed:
2.3.0

vademo committed:
iam-aprofiel-phone method

Jan-Bart committed:
Merge pull request #53 from digipolisantwerp/feature/v2AddAuthMethod

iam-aprofiel-phone method

Jan-Bart committed:
2.3.1

Jan-Bart committed:
Update package.lock + add dist folder for release

Jan-Bart committed:
Added dist

Jan-Bart committed:
🔧 Added publishConfig to prevent publishing to internal registry

Jan-Bart committed:
Fix auth_method astad.aprofiel.v1

Jan-Bart committed:
2.3.3

Jan-Bart committed:
Added test and documentation upgrade

Jan-Bart committed:
2.3.4

Jan-Bart committed:
Merge pull request #55 from digipolisantwerp/bug/ACAPP-434

Bug/acapp 434

horstenwillem committed:
chore: update dependencies + add lintr + add github workflows

horstenwillem committed:
add Typescript types

horstenwillem committed:
add .github folder to npmignore

Jan-Bart committed:
Merge pull request #61 from horstenwillem/chore/maintenance

chore/maintenance

osamah-dgp committed:
Add configurable fromUrl domain check

osamah-dgp committed:
Fix syntax error in readme docs

osamah-dgp committed:
Fix linting issues

osamah-dgp committed:
Add test checking fromUrl logic, fix existing tests, add invalid url fallback

Jan-Bart committed:
Merge pull request #72 from osamah-dgp/ASTAD-28445-open-redirect-issue-na-auth

ASTAD-28445 Open redirect issue na auth

Jan-Bart committed:
2.4.0

Jan-Bart committed:
Updated changelog

osamah-dgp committed:
Fix ts issues

osamah-dgp committed:
Small lint fix

osamah-dgp committed:
Add minimum node 15 requirement

Jan-Bart committed:
Merge pull request #73 from osamah-dgp/ASTAD-28445-open-redirect-issue-na-auth

ASTAD-28445 open redirect issue na auth

Jan-Bart committed:
Updated lock-file

Jan-Bart committed:
2.4.1

osamah-dgp committed:
Add crash handler to failed refresh calls and additional error logging

osamah-dgp committed:
Lint fixes

Jan-Bart committed:
Merge pull request #74 from osamah-dgp/ASTAD-28488-crash-fixes

Add crash handler to failed refresh calls and additional error logging

Jan-Bart committed:
2.4.2

Commit:
Fix ASTAD-28929 open redirect

Commit:
2.4.3-0

Commit:
Fix open redirect on logout for users without session

Commit:
2.4.3-1

Commit:
2.5.0
